### PR TITLE
Setup/interviews

### PR DESCRIPTION
**Pull request by Philipp101**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class User < ApplicationRecord
-  has_many :bookings
-  has_many :interviews, through: :bookings
+  has_many :bookings, dependent: :destroy
+  has_many :interviews, through: :bookings, dependent: :destroy
   validates :nickname, uniqueness: true
```
Comment (mstehling) [inline app/models/user.rb:5]: I think we also need to add `has_many :interviews, dependent: :destroy` (i.e. without the "through bookings"), to account for the interviews **created** by the user. We need to ask a TA tomorrow how rails then knows what we want when we say user1.interviews (i.e. do we mean those interviews user1 joined or those user1 created).

File: app/controllers/interviews_controller.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+class InterviewsController < ApplicationController
+  #skip_before_action :authenticate_user!, only: [:index, :show]
```
Comment (mstehling) [inline app/controllers/interviews_controller.rb:2]: I understand the interview :show view is the one we see after clicking on "Book" in the : index, correct?
If that is the case, then we don't have a view yet for when the users conduct the interview (i.e. video, chat, etc. - see Figma) - we need to add another route for this. Let's discuss how we name the route.

File: app/models/user.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 class User < ApplicationRecord
-  has_many :bookings
-  has_many :interviews, through: :bookings
+  has_many :bookings, dependent: :destroy
+  has_many :interviews, through: :bookings, dependent: :destroy
```
Comment (mstehling) [inline app/models/user.rb]: I think for `has_many :interviews, through: :bookings, dependent: :destroy`, the `dependent: :destroy` needs to be removed. What we say here is that when we delete user1, then we want to delete all interviews (which other users created) that he booked. So in effect, we are deleting other users' interview sessions.

File: app/views/interviews/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+              <%= interview.time.strftime("%a %e.%m.%y | %l:%M") %>
+            </div>
+          </div>
+
+          <div>
+            <div>
+              <!-- Icon Code -->
+            </div>
+            <div>
+              <!-- Tags -->
```
Comment (mstehling) [inline app/views/interviews/index.html.erb:20]: For today's demo, should we just add the tags as a list here?

File: app/controllers/interviews_controller.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    @interview = Interview.new
+    #authorize @interview
+  end
+
+  def create
+    @interview = Interview.new(interview_params)
+    #authorize @interview
+    @interview.user = current_user
+    if @interview.save
+      redirect_to interview_path(@interview)
```
Comment (mstehling) [inline app/controllers/interviews_controller.rb]: I believe we would redirect to the dashboard, not the interview :show view.

File: db/seeds.rb [hunk 6]
```diff
@@ -5,3 +5,37 @@
...
+interview6 = Interview.create!(time: DateTime.new(2020,12,12,18), focus: "IT Consultant", experience: "Senior level", interview_language: "dutch", user: bruce)
+interview7 = Interview.create!(time: DateTime.new(2009,9,1,17), focus: "Ruby on Rails developer", experience: "expert", interview_language: "french", user: john)
+interview8 = Interview.create!(time: DateTime.new(2020,12,12,17), focus: "Head of Development", experience: "medium", interview_language: "english", user: mary)
+interview9 = Interview.create!(time: DateTime.new(2020,12,12,18), focus: "Frontend WebDev", experience: "Senior level", interview_language: "german", user: zoe)
+interview10 = Interview.create!(time: DateTime.new(2009,9,1,17), focus: "PHP developer", experience: "beginner", interview_language: "english", user: peter)
+interview11 = Interview.create!(time: DateTime.new(2020,12,12,17), focus: "PHP", experience: "medium", interview_language: "english", user: lisa)
+interview12 = Interview.create!(time: DateTime.new(2020,12,12,18), focus: "General IT Job", experience: "Senior level", interview_language: "spanish", user: bruce)
```
Comment (mstehling) [inline db/seeds.rb]: The focus attribute will contain our tags in the end, i.e. Ruby, CSS, JS, HTML. For the demo I suggest we implement this in the seed by just assigning 1 of these 4 as focus.

**Review by mstehling — COMMENTED** [review]
Great work, thanks! Let's discuss the comments if there are any questions :)

File: app/controllers/interviews_controller.rb (hunk 2 above)
Comment (Philipp101) [reply app/controllers/interviews_controller.rb:2]: or we integrate the video frame (and the chat frame) on that page as I suggested yesterday and make it some kind of blocked/greyish until the interview time frame starts...

File: app/views/interviews/index.html.erb (hunk 4 above)
Comment (Philipp101) [reply app/views/interviews/index.html.erb:20]: OK.

File: app/controllers/interviews_controller.rb (hunk 5 above)
Comment (Philipp101) [reply app/controllers/interviews_controller.rb]: true

File: db/seeds.rb (hunk 6 above)
Comment (Philipp101) [reply db/seeds.rb]: easy change. will be done!

**Pull request merged**